services/blockchain-indexer/shared/dataService/business/interoperability/blockchainApps.js

```diff
@@ -44,11 +46,34 @@ const getBlockchainApps = async (params) => {
 
 	const total = await blockchainAppsDB.count(params);
 
-	blockchainAppsInfo.data = await blockchainAppsDB.find(
+	const response = await blockchainAppsDB.find(
 		{ ...params, limit: params.limit || total },
 		Object.getOwnPropertyNames(blockchainAppsIndexSchema.schema),
 	);
 
+	blockchainAppsInfo.data = await BluebirdPromise.map(
+		response,
+		async (appInfo) => {
+			if (!appInfo.isDefault) {
+				const isDefault = config.defaultApps.some(e => e === appInfo.name);
+
+				const blockchainAppInfo = {
+					...appInfo,
+					isDefault,
+				};
+
+				if (isDefault) await blockchainAppsDB.upsert(blockchainAppInfo);
```

Comment:
We do not need to wait for the index to be updated for the request to be served. This can happen concurrently while the user request is already served.
```suggestion
				if (isDefault) blockchainAppsDB.upsert(blockchainAppInfo);
```